### ISSUE TITLE
Improve documentation on Gradle build cache timeout

### ISSUE THREAD

**Issue by jfbibeau** [gradle/gradle#5520]
Opening an issue as requested by @lptr 

### Expected Behavior
I'd love to see a bit more documentation around the build cache timeout. I see it was added in Gradle 4.3 (https://docs.gradle.org/4.3/release-notes.html#timeouts-for-http/https-requests) however there's no mention of:

1. What the default values are, and;
2. How to modify them.

### Current Behavior
I'm kind of able to infer this information by looking at the GitHub PRs which introduced this feature, but that's not ideal. Maybe an opportunity to improve the docs in this area.

I'd also like to understand the difference between the 2 timeouts configurable, with the intent of answering the question "If I have a really slow connection and downloading the artifact takes a loooong time, will either of these timeout interrupt the download, or will these only address unreachable remote caches?"

### Context
Some of our users are experiencing long-running download tasks from the Gradle Enterprise remote build cache and we'd like to understand how we can tweak Gradle cache timeouts.

### Steps to Reproduce (for bugs)
N/A

### Your Environment
Gradle 4.7, Gradle Enterprise 2018.2.1

Thank you!

**Comment by lptr** [comment]
See also gradle/gradle#3345.

**Comment by Charis1331** [comment]
@lptr Is adding a custom timeout feature considered in your roadmap?

**Comment by lptr** [comment]
@Charis1331 do you mean a separate timeout for only the HTTP cache, but not for the dependency downloads? What is your use case where such a feature would be useful?

**Comment by Charis1331** [comment]
@lptr Yeah (sorry for not specifying it), I was referring to the HTTP cache as stated, also, in the [#3345](https://github.com/gradle/gradle/issues/3345). I saw the [workaround](https://github.com/gradle/gradle/issues/3345#issuecomment-346027369) mentioned in the above issue, but I was thinking if it would be possible to ping the remote cache's URL, even if **you are connected** to your company's VPN, and if the timeout is reached go ahead and disable it. My question lines up pretty much with the issue I referenced.

**Comment by ericoporto** [comment]
> I was referring to the HTTP cache as stated, also, in the https://github.com/gradle/gradle/issues/3345. I saw the https://github.com/gradle/gradle/issues/3345#issuecomment-346027369 mentioned in the above issue, but I was thinking if it would be possible to ping the remote cache's URL, even if you are connected to your company's VPN, and if the timeout is reached go ahead and disable it.

That would be pretty sweet!
